File: go-service/models/customer.go
package models

import "time"

// Customer represents a customer in the system
type Customer struct {
	ID           int        `db:"Id" json:"id"`
	Name         string     `db:"Name" json:"name"`
	Phone        *string    `db:"Phone" json:"phone,omitempty"`
	Email        *string    `db:"Email" json:"email,omitempty"`
	Address      *string    `db:"Address" json:"address,omitempty"`
	CustomerType string     `db:"CustomerType" json:"customerType"`
	IsActive     bool       `db:"IsActive" json:"isActive"`
	CreatedAt    time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt    *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: backend/Api/Models/Customer.cs
using System.Text.Json.Serialization;
using Api.Enums;

namespace Api.Models
{
    public class Customer
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }
        public CustomerType CustomerType { get; set; } = CustomerType.Individual;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [JsonIgnore]
        public virtual ICollection<Sale> Sales { get; set; } = new List<Sale>();
        [JsonIgnore]
        public virtual ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();
    }
}
